src/App.tsx

```diff
@@ -1,61 +1,122 @@
+import { ChangeEvent, MouseEvent, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const initialTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(initialTodos);
+  const [title, setTitle] = useState('');
+  const [isValidTitle, setIsValidTitle] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [isValidUser, setIsValidUser] = useState(false);
+
+  let count = 15;
+
+  const handleSelect = (event: ChangeEvent<HTMLSelectElement>) => {
+    setIsValidUser(false);
+    setUserId(+event.target.value);
+  };
+
+  const handleTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setIsValidTitle(false);
+    setTitle(event.target.value);
+  };
+
+  const createTodo = (event: MouseEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setIsValidTitle(true);
+    }
+
+    if (!userId) {
+      setIsValidUser(true);
+    }
+
+    if (!title.trim() || !userId) {
+      return;
+    }
```

Comment:
```suggestion
    if (!title.trim()) {
      return setIsValidTitle(true);
    }

    if (!userId) {
      return setIsValidUser(true);
    }
```

Reply:
it's don't work great. If you don't change user and title, you have to see two errors: "Please enter a title" and "Please choose a user". Your code don't complete this task.